File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int ans=helper("",s);
        return ans;

    }

    static int helper(String p, String up){
        if(up.isEmpty()){

            return p.length();
        }
        char c = up.charAt(0);
        boolean ch=true;
        for(int i=0;i<p.length();i++){
            if(c==p.charAt(i)){
                ch=false;
                break;
            }
        }
        if(ch==false){
            return helper(p,up.substring(1));
        }else{
            return helper(p+c,up.substring(1));
        }

    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> mp;
        int left=0;
        int ans=0;
        for(int right=0;right<s.length();right++){
            char c=s[right];
            mp[c]++;
            while(mp[c]>1){
                mp[s[left]]--;
                left++;
            }
            ans=max(ans,right-left+1);
        }
        return ans;
    }
};